src/main/java/at/ac/tuwien/damap/rest/persons/orcid/ORCIDMapper.java

```diff
@@ -0,0 +1,121 @@
+import at.ac.tuwien.damap.rest.persons.orcid.models.base.ORCIDDate;
+import at.ac.tuwien.damap.rest.persons.orcid.models.base.ORCIDEmail;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class ORCIDMapper {
+    public ContributorDO mapExpandedSearchPersonEntityToDO(ORCIDExpandedSearchPerson orcidPerson,
+            ContributorDO contributorDO) {
+
+        contributorDO.setId(null);
+        contributorDO.setFirstName(orcidPerson.getGivenNames());
+        contributorDO.setLastName(orcidPerson.getFamilyNames());
+
+        String firstMail = orcidPerson.getEmails().isEmpty() ? null : orcidPerson.getEmails().get(0);
+        contributorDO.setMbox(firstMail);
+
+        String firstAffiliation = orcidPerson.getAffiliations().isEmpty() ? null
+                : orcidPerson.getAffiliations().get(0);
+        contributorDO.setAffiliation(firstAffiliation);
+
+        IdentifierDO identifierContributorDO = new IdentifierDO();
+        identifierContributorDO.setIdentifier(orcidPerson.getOrcidId());
+        identifierContributorDO.setType(EIdentifierType.ORCID);
+
+        contributorDO.setPersonId(identifierContributorDO);
+
+        return contributorDO;
+    }
+
+    public ContributorDO mapRecordEntityToPersonDO(ORCIDRecord orcidRecord,
+            ContributorDO contributorDO) {
+
+        contributorDO.setId(null);
+        contributorDO.setFirstName(orcidRecord.getPerson().getName().getGivenNames().getValue());
+        contributorDO.setLastName(orcidRecord.getPerson().getName().getFamilyName().getValue());
+
+        var primaryMail = orcidRecord.getPerson().getEmails().getEmail().stream().filter(ORCIDEmail::isPrimary)
+                .findFirst();
+        contributorDO.setMbox(primaryMail.isPresent() ? primaryMail.get().getEmail() : null);
+
+        List<ORCIDAffiliation> affiliations = new ArrayList<>();
+
+        for (var groupListItem : List.of(
+                orcidRecord.getActivitiesSummary().getEducations().getAffiliationGroup(),
+                orcidRecord.getActivitiesSummary().getEmployments().getAffiliationGroup())) {
+
+            for (var group : groupListItem) {
+
+                for (var summary : group.getSummaries()) {
+                    affiliations.add(summary.getSummary());
+                }
+            }
+
+        }
+
+        if (!affiliations.isEmpty()) {
+            affiliations.sort(sortByCurrentStartDate);
+            contributorDO.setAffiliation(affiliations.get(affiliations.size() - 1).getOrganization().getName());
+        }
+
+        IdentifierDO identifierContributorDO = new IdentifierDO();
+        identifierContributorDO.setIdentifier(orcidRecord.getPerson().getName().getPath());
+        identifierContributorDO.setType(EIdentifierType.ORCID);
+
+        contributorDO.setPersonId(identifierContributorDO);
+        return contributorDO;
+    }
+
+    private final Comparator<? super ORCIDAffiliation> sortByCurrentStartDate = ((ORCIDAffiliation a,
```

Comment:
The ORCIDRecord does not contain this information, but maybe there is an easy way to fetch the source of an affiliation and give preferential treatment to those that were not created by the orcid person. See the picture, right now DAMAP chooses "Interdisziplinäre Werteforschung"  instead of "Universität Wien" as the affiliation. Just a suggestion, if it is easily doable.
<img width="748" alt="image" src="https://github.com/tuwien-csd/damap-backend/assets/80054001/c7059206-4bd6-41c6-9bb5-c9643ceb5e57">